File: in_C++/1900A_cover_in_water.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int num;
        cin >> num;
        string str;
        cin >> str;
        int min = 0;

        for (int i = 2; i < num; i++)
        {
            if (str[i] == '.' and str[i - 1] == '.' and str[i - 2] == '.')
            {
                min = 2;
            }
        }
        if (min < 2)
        {

            for (int i = 0; i < num; i++)
            {
                if (str[i] == '.') {
                    min= min +1;
                }
            }
        }
        cout << min << endl;
    }
    return 0;
}

File: in_Python/1900A_cover_in_water.py
t = int(input())
for i in range(t):
    num = int(input())
    lol = str(input())
    min = 0;
    
    for _ in range(2,num):
        if lol[_-2] == '.' and lol[_-1] == '.' and lol[_] == '.':
            min = 2;
            
    if min < 2:
        for  _ in range(num):
            if lol[_] == '.':
                min = min + 1
    print(min)